src/packages/utils/src/utils/random-number-generator.ts

```diff
@@ -0,0 +1,37 @@
+import seedrandom from "seedrandom";
+
+export class RandomNumberGenerator {
+  readonly rng: () => number;
+
+  constructor(seed?: string) {
+    if (seed) {
+      this.rng = seedrandom.alea(seed);
+    } else {
+      this.rng = Math.random;
+    }
+  }
+
+  getNumber(range: number = 1): number {
+    // I believe this check may be a tiny bit faster than
+    // always multiplying by 1
+    if (range !== 1) {
+      return this.rng() * range;
```

Comment:
While reading this I noticed that the range of number is exclusive of the `range` value as the rng outputs values from `[0 to 1)` (0 inclusive, 1 exclusive), so when multiplied by `255` the max value is `255.9999...`. `Buffer.from` will automatically case these values to an integer by lopping of the decimal (`Buffer.from([254.999]) -> `[fe]`).

The implementation in ethereum has the same bug. I'll fix the little purpose-built RNG used over there; probably by increasing the upper range by 1. I dunno, haven't thought about it yet.

Reply:
I guess `range` is a misnomer here because of that issue; perhaps it should be named `upperExclusiveBound`?

I believe that we shouldn't try to manipulate it to output a `[0 to 1]` set of values, but rather change the consuming code if it needs to make sure the upper bound is met sometimes. For example, if we're using `Buffer.from([rng.getNumber(255)])` to generate an address which we know that we sometimes want `ff`, then we can do `Buffer.from([Math.round(rng.getNumber(255))])` which is safe because we know `Buffer.from` is already going to truncate it, so that is perfectly valid code

Otherwise we can add a method here for `getRoundedInteger(range: number) : number` which calls the `Math.round` for the tool. However, we can't do this in `getNumber` because it's hard to know what this random number is being used for, and whether or not it's range is large enough to be rounded

An interesting issue that I didn't consider. Let me know if/how/when you'd like this to be changed

Reply:
Here is my super simple fix: https://github.com/trufflesuite/ganache-core/pull/770

I think just renaming it `upperExclusiveBound` would be fine here. Then just update `getBuffer`'s `range` to be `256`. I think that'll work ok here. We don't need a perfectly random distribution, but i'd like to be able to get all numbers between 0 and 255, inclusive of both.